[glozxi] iP

glozxi opened the pull request:
# Duke ❤️
> "I love Duke!" - me

Features:
- **Add** tasks
- **Delete** tasks
- **Mark** tasks as done/undone
- ~~Love Duke~~

But where's Duke?
1. Download [here](https://github.com/glozxi/ip/releases/tag/v0.1).
2. Open it with this command: `java -jar duke.jar`.

Stuff I still have to do:
- [x] Love Duke
- [ ] Love Duke
- [ ] Love Duke

Stuff you have to do:
```java
public class Main {
    public static void main(String[] args) {
        System.out.println("I love Duke.");
    }
}
```

Commit:
Add Gradle support

glozxi committed:
Modify greeting

glozxi committed:
Add buffered reader

glozxi committed:
Add format to Duke's messages

glozxi committed:
Add user input

Replace buffered reader with scanner. User can exit by inputting "exit".

glozxi committed:
Change exit command to 'bye"

glozxi committed:
Add echo to user input

glozxi committed:
Move printed strings into methods

glozxi committed:
Add Task.java and TaskList.java

glozxi committed:
Add storing and displaying task

glozxi committed:
Remove unneeded 'this' keywords

glozxi committed:
Remove last new line from TaskList string

glozxi committed:
Add display of the added task

glozxi committed:
Add methods to mark tasks as done or undone

glozxi committed:
Add ability to mark tasks as done or undone

glozxi committed:
Add text display after mark, unmark and list

glozxi committed:
Add deadline, event, todo classes

glozxi committed:
Add ability to track todos, events, deadlines

glozxi committed:
Change task list for adding tasks directly

Pass in tasks instead of strings

glozxi committed:
Add automated text UI testing

glozxi committed:
Add DukeException.java

glozxi committed:
Add invalid command exception

glozxi committed:
Add invalid argument exception

glozxi committed:
Add empty argument exception

glozxi committed:
Add empty argument exception and exception tests

glozxi committed:
Add methods to remove task

glozxi committed:
Change checking of invalid task number

glozxi committed:
Add ability to delete tasks

glozxi committed:
Use enums for command types

glozxi committed:
Add save and load methods

glozxi committed:
Change save and load methods to throw exceptions

glozxi committed:
Use LocalDateTime for date and time

glozxi committed:
Merge branch 'branch-Level-7'

glozxi committed:
Merge branch 'branch-Level-8'

# Conflicts:
#	src/main/java/Duke.java

glozxi committed:
Merge remote-tracking branch 'upstream/add-gradle-support'

glozxi committed:
Add Storage and Ui classes

glozxi committed:
Correct merge mistakes

glozxi committed:
Use Ui for interactions with the user

glozxi committed:
Fix saving and loading date and time

glozxi committed:
Move save and load methods to Storage class

glozxi committed:
Remove unused method

glozxi committed:
Use instance methods and variables for Duke.java

glozxi committed:
Add a few command classes

glozxi committed:
Add DeleteTaskCommand class

glozxi committed:
Add MarkTaskAsDone class

glozxi committed:
Add error throw to deleteTask method

Fix tests.

glozxi committed:
Add MarkTaskAsUndoneCommand class

glozxi committed:
Add command classes for adding tasks

glozxi committed:
Use Parser to parse commands

glozxi committed:
Move date time formatter for saving to storage

glozxi committed:
Divide classes into packages

glozxi committed:
Add JUnit tests for ToDoTask and Ui

glozxi committed:
Change main class name in build.gradle

glozxi committed:
Add JavaDoc comments to Duke.java

glozxi committed:
Add JavaDoc comments to Ui.java

glozxi committed:
Add JavaDoc comments

glozxi committed:
Edit TaskList.java

glozxi committed:
Add JavaDoc comments to TaskList.java

glozxi committed:
Add JavaDoc comments

glozxi committed:
Remove unused components from CommandType.java

glozxi committed:
Change code to comply with coding standard

glozxi committed:
Add find command

glozxi committed:
Add ability to parse find command

glozxi committed:
Add tests for find in text-ui-test

glozxi committed:
Merge branch 'branch-A-JavaDoc'

glozxi committed:
Merge branch 'branch-A-CodingStandard'

glozxi committed:
Merge branch 'branch-Level-9'

# Conflicts:
#	src/main/java/duke/commands/CommandType.java
#	src/main/java/duke/task/Task.java

glozxi committed:
Correct coding style violations using checkStyle

glozxi committed:
Merge branch 'branch-A-CheckStyle'

glozxi committed:
Add GUI

glozxi committed:
Merge branch 'branch-Level-10'

glozxi committed:
Change Ui class

Rename it, remove text tests and change unit tests

glozxi committed:
Add greeting on launch

glozxi committed:
Change code to match coding standards

dillongoh commented on src/main/java/duke/Duke.java:
```diff
@@ -0,0 +1,57 @@
+    /**
+     * You should have your own function to generate a response to user input.
+     * Replace this stub with your completed method.
+     */
```
This javadocs is from the guide. Perhaps you can modify it to for your own code 😄

dillongoh commented on src/main/java/duke/commands/Command.java:
```diff
@@ -0,0 +1,25 @@
+     * @param uiText The ui to be used.
+     * @param storage The storage to be used.
+     * @return The string to be printed.
+     * @throws DukeException If the command cannot be executed correctly.
```
I like how your javadocs are very detailed for all the methods I've seen! Good job!

dillongoh commented on src/main/java/duke/commands/CommandType.java:
```diff
@@ -0,0 +1,35 @@
+    }
+
+    public static CommandType getCommandType(String command) throws DukeException {
+        CommandType[] values = values();
```
I like how you attached the array specifier to the type instead of the variable

dillongoh commented on src/main/java/duke/task/TaskList.java:
```diff
@@ -0,0 +1,107 @@
+        return list.get(getIndexFromNumber(number)).toString();
+    }
+
+    public TaskList getMatchingTasks(String keyword) {
```
This method could use some javadocs to tell readers what the method does

dillongoh reviewed (commented):
Overall I found your code very clear with very good adherence to the coding standards. Very detailed Javadocs for almost all public methods too! 👍

TimothyLawSongEn commented on src/main/java/duke/task/ToDoTask.java:
```diff
@@ -0,0 +1,18 @@
+ */
+public class ToDoTask extends Task {
+
+    private static final String EVENT_SYMBOL = "T";
```
I like how you created a constant for EVENT_SYMBOL instead of hard-coding into you methods.

TimothyLawSongEn reviewed (commented):
LGTM. Only a few nits to fix. Your code quality is decent overall, with no methods being unnecessarily difficult to understand.

glozxi committed:
Update fxml files

glozxi committed:
Update Javadocs

glozxi committed:
Add assertions

glozxi committed:
Improve code quality

glozxi committed:
Merge pull request #1 from glozxi/branch-A-Assertions

Add assertions

glozxi committed:
Merge branch 'master' into branch-A-CodeQuality

# Conflicts:
#	src/main/java/duke/parser/Parser.java

glozxi committed:
Merge pull request #2 from glozxi/branch-A-CodeQuality

Improve code quality

glozxi committed:
Add .yml file

glozxi committed:
Add reminder command

There is no reminder command for users to see upcoming deadlines and
events.

A reminder command is useful for users to check what to do soon, so that
they do not have to check the whole list of tasks.

Let's add a reminder command that will display the deadlines that end
and events that start within 2 days from the time that the command is
entered. A todo will not be reminded about as it does not have a date
that it needs to be finished by.

glozxi committed:
Optimize imports

glozxi committed:
Add a comment

glozxi committed:
Remove unnecessary throws

glozxi committed:
Fix coding standard violations

glozxi committed:
Change UI

The text is slanted, and can be blocked by text input box when the
text is too long.

Straightening it can allow long text to be viewed.

Let's straighten the label that has the text.

It is preferable to allow all text to be seen if the user adds many
tasks with long descriptions and displays them.

glozxi committed:
Merge branch 'branch-BCD-Extension'

glozxi committed:
Edit Javadocs

glozxi committed:
Update UI

glozxi committed:
Update reminder command

glozxi committed:
Update input textbox color

glozxi committed:
Add tests

glozxi committed:
Change personality

glozxi committed:
Add response when no tasks are found for find command

glozxi committed:
Add response when there are no tasks for list command

glozxi committed:
Change conditions for reminders

A done event or deadline can be shown in reminders. A event or deadline
less than 1 day from now is not shown in reminders.

Changing them makes Duke's reminders more convenient.

Let's
* Not show events or deadlines marked as done in reminders
* Show events and deadlines less than 1 day from now in reminders

glozxi committed:
Add title to window

glozxi committed:
Fix save after every command

glozxi committed:
Rename duke to angela

glozxi committed:
Add Ui.png

glozxi committed:
Add a User Guide

glozxi committed:
Use GFM

glozxi committed:
Add icon

glozxi committed:
Update Ui.png

glozxi committed:
Change fxml javafx version to 11

glozxi committed:
Improve code quality

glozxi committed:
Update User Guide

glozxi committed:
Update User Guide

glozxi committed:
Remove image from User Guide

glozxi committed:
Edit README.md to be user guide

glozxi committed:
Use Arial font instead of Gill Sans MT
